File: Plateform_BackEnd/src/main/java/com/tourist/model/Room.java
package com.tourist.model;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.tourist.enums.Type;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;


@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Room {
     @Id
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private Type type;
     private Long price;
     private boolean available;

     @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
     private List<Image> images;


     @ManyToOne()
     @JoinColumn(name = "hotel_id")
     @JsonIgnore
     private Hotel hotel;

     @OneToMany(mappedBy = "room" , cascade = CascadeType.ALL , orphanRemoval = true)
     @JsonIgnore
     private List<Reservation> reservations = new ArrayList<>();



}

File: Plateform_FrontEnd/src/app/model/room/room.ts
import {Reservation} from "../reservation/reservation";
import {Type} from "../../enums/type";
import {Image} from "../image/image";
import {Hotel} from "../hotel/hotel";

export class Room {
  id !: number;
  type!: Type;
  price!: number;
  available!: boolean;
  images: Image[] = [];
  hotel: Hotel;
  reservations: Reservation[];


}
